lib/xctest.js

```diff
@@ -0,0 +1,180 @@
+        const xctestConfiguration = new XCTestConfiguration({
+            testBundleURL: `file://${appInfo.Path}/PlugIns/${targetName}.xctest`,
+            sessionIdentifier,
+            targetApplicationBundleID: this.targetBundleId,
+        });
+        const xctestContent = xctestConfiguration.getBytes();
+        const houseArrestService = await startHouseArrestService(this.udid);
+        const vendContainer = await houseArrestService.vendContainer(this.xctestBundleId);
+        const list = await vendContainer.listDirectory('/tmp');
+        for (let file of list) {
+            if (file.endsWith('.xctestconfiguration')) {
+                log.debug('removing /tmp/' + file);
+                await vendContainer.deleteDirectory('/tmp/' + file);
+            }
+        }
+        const stream = await vendContainer.createWriteStream(xctestPath, {});
+        await new B((resolve, reject) => {
+            stream.write(xctestContent, resolve);
+            stream.on('error', reject);
+        });
+        stream.end();
+        vendContainer.close();
+        houseArrestService.close();
+        const instrumentService = await startInstrumentService(this.udid);
+        const instrumentDaemon = await instrumentService.makeChannel(PROCESS_CONTROL);
+        await instrumentService._callChannel(true, instrumentDaemon, 'processIdentifierForBundleIdentifier:', this.xctestBundleId);
+        const appPath = appInfo.Path;
+        const xctestConfigurationPath = appContainer + xctestPath;
+        const appEnv = {
+            CA_ASSERT_MAIN_THREAD_TRANSACTIONS: '0',
+            CA_DEBUG_TRANSACTIONS: '0',
+            DYLD_FRAMEWORK_PATH: appPath + '/Frameworks:',
+            DYLD_LIBRARY_PATH: appPath + '/Frameworks',
+            NSUnbufferedIO: 'YES',
+            SQLITE_ENABLE_THREAD_ASSERTIONS: '1',
+            WDA_PRODUCT_BUNDLE_IDENTIFIER: '',
+            XCTestConfigurationFilePath: xctestConfigurationPath,
+            XCODE_DBG_XPC_EXCLUSIONS: 'com.apple.dt.xctestSymbolicator',
+            MJPEG_SERVER_PORT: '',
+            USE_PORT: '',
+            LLVM_PROFILE_FILE: appContainer + '/tmp/%p.profraw',
+        };
+        if (majorVersion >= 11) {
+            appEnv.DYLD_INSERT_LIBRARIES = '/Developer/usr/lib/libMainThreadChecker.dylib';
+            appEnv.OS_ACTIVITY_DT_MODE = 'YES';
+        }
+        const appArgs = [
+            '-NSTreatUnknownArgumentsAsOpen', 'NO',
+            '-ApplePersistenceIgnoreState', 'YES'
+        ];
+        const appOptions = { StartSuspendedKey: false };
+        if (majorVersion >= 12) {
+            appOptions.ActivateSuspended = true;
+        }
+        const pid = await instrumentService._callChannel(true, instrumentDaemon,
+            'launchSuspendedProcessWithDevicePath:bundleIdentifier:environment:arguments:options:',
+            appPath, this.xctestBundleId, appEnv, appArgs, appOptions);
+        log.info(`Pid of launched ${this.xctestBundleId} ${pid.selector}:`);
```

Comment:
Here can call `callChannel` like https://github.com/appium/appium-ios-device/blob/6ee093956bbf0f3f1f332095e7eff2bff70c269f/test/instrument/instrument-service-specs.js#L28-L29
instead of `_callChannel`

Reply:
solved